controllers/runner_controller.go

```diff
@@ -234,6 +259,9 @@ func (r *RunnerReconciler) unregisterRunner(ctx context.Context, org, repo, name
 	id := int64(0)
 	for _, runner := range runners {
 		if runner.GetName() == name {
+			if runner.GetBusy() {
```

Comment:
Good to know that there is an official API to determine if a runner is busy or not 😃

Reply:
That's right 🙂 I have closed the PR for now because I am still testing my changes and adding a few more. I will reopen it when I finish